pkgs/applications/science/robotics/mujoco/default.nix

```diff
@@ -1,122 +1,236 @@
-  src = fetchFromGitHub {
-    owner = "deepmind";
-    repo = pname;
-    rev = version;
-    hash = "sha256-LgpA+iPGqciHuWBSD6/7yvZ7p+vo48ZYKjjrDZSnAwE=";
   };
 
-  patches = [ ./dependencies.patch ];
-
-  nativeBuildInputs = [ cmake git ];
-
-  buildInputs = [
-    libGL
-    xorg.libX11
-    xorg.libXcursor
-    xorg.libXext
-    xorg.libXi
-    xorg.libXinerama
-    xorg.libXrandr
-  ];
-
-  # Move things into place so that cmake doesn't try downloading dependencies.
-  preConfigure = ''
-    mkdir -p build/_deps
-    ln -s ${abseil-cpp} build/_deps/abseil-cpp-src
-    ln -s ${benchmark} build/_deps/benchmark-src
-    ln -s ${ccd} build/_deps/ccd-src
-    ln -s ${eigen3} build/_deps/eigen3-src
-    ln -s ${glfw3} build/_deps/glfw3-src
-    ln -s ${googletest} build/_deps/googletest-src
-    ln -s ${lodepng} build/_deps/lodepng-src
-    ln -s ${qhull} build/_deps/qhull-src
-    ln -s ${tinyobjloader} build/_deps/tinyobjloader-src
-    ln -s ${tinyxml2} build/_deps/tinyxml2-src
-  '';
-
-  meta = with lib; {
-    description = "Multi-Joint dynamics with Contact. A general purpose physics simulator.";
-    homepage = "https://mujoco.org/";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ samuela ];
+  mujoco = stdenv.mkDerivation rec {
+    inherit pname version;
+
+    src = fetchFromGitHub {
+      owner = "google-deepmind";
+      repo = pname;
+      rev = version;
+      hash = "sha256-zHK1uYtWkKFWBMqkw/NwuBlLoXyiZJeNC/sSXwSfgW4=";
+    };
+
+    patches = [ ./mujoco-system-deps-dont-fetch.patch ];
+
+    nativeBuildInputs = [ cmake ];
+
+    buildInputs = [
+      pin.sdflib
+      glm
+
+      # non-numerical
+      spdlog
+      cereal_1_3_2
+      glfw
+    ];
+
+    cmakeFlags = [
+      "-DMUJOCO_USE_SYSTEM_sdflib=ON"
+      "-DMUJOCO_SIMULATE_USE_SYSTEM_GLFW=ON"
+      "-DMUJOCO_SAMPLES_USE_SYSTEM_GLFW=ON"
+    ];
+
+    # Move things into place so that cmake doesn't try downloading dependencies.
+    preConfigure = ''
+      mkdir -p build/_deps
+      ln -s ${pin.abseil-cpp} build/_deps/abseil-cpp-src
+      ln -s ${pin.benchmark} build/_deps/benchmark-src
+      ln -s ${pin.ccd} build/_deps/ccd-src
+      ln -s ${pin.eigen3} build/_deps/eigen3-src
+      ln -s ${pin.googletest} build/_deps/googletest-src
+      ln -s ${pin.lodepng} build/_deps/lodepng-src
+      ln -s ${pin.qhull} build/_deps/qhull-src
+      ln -s ${pin.tinyobjloader} build/_deps/tinyobjloader-src
+      ln -s ${pin.tinyxml2} build/_deps/tinyxml2-src
+      ln -s ${pin.marchingcubecpp} build/_deps/marchingcubecpp-src
+    '';
+
+    meta = with lib; {
+      description =
+        "Multi-Joint dynamics with Contact. A general purpose physics simulator.";
+      homepage = "https://mujoco.org/";
+      license = licenses.asl20;
+      maintainers = with maintainers; [ samuela tmplt ];
+    };
+  };
+
+  python-bindings = python3Packages.buildPythonPackage rec {
+    inherit pname version;
+    format = "pyproject";
+
+    src = fetchPypi {
```

Comment:
why not `fetchFromGitHub`?

Reply:
Because the PyPi tarball is impurely built via [`python/make_sdist.sh`](https://github.com/google-deepmind/mujoco/blob/main/python/make_sdist.sh) in the project's CI. I suppose it can be patched, but it was nothing I wanted to handle this first pass.

Reply:
Ah ok thanks for clarifying! in that case, let's leave a code comment explaining this situation